src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> l = last;
+        final Node<T> newNode = new Node<>(l, value, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else {
+            checkIndex(index);
+            Node<T> pred = node(index).prev;
+            Node<T> newNode = new Node<>(pred, value, node(index));
+            node(index).prev = newNode;
+            if (pred == null) {
+                first = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            pred = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t: list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> x = node(index);
```

Comment:
```suggestion
        Node<T> node = node(index);
```
https://google.github.io/styleguide/javaguide.html#s5.2.5-non-constant-field-names